awx/main/models/notifications.py

```diff
@@ -233,14 +237,20 @@ def send_notification_templates(self, status_str):
         if notification_templates:
             if status_str == 'succeeded':
                 notification_template_type = 'success'
+            elif status_str == 'running':
+                notification_template_type = 'started'
             else:
                 notification_template_type = 'error'
-            all_notification_templates = set(notification_templates.get(notification_template_type, []) + notification_templates.get('any', []))
-            if len(all_notification_templates):
-                try:
-                    (notification_subject, notification_body) = getattr(self, 'build_notification_%s_message' % status_str)()
-                except AttributeError:
-                    raise NotImplementedError("build_notification_%s_message() does not exist" % status_str)
+            all_notification_templates = set(notification_templates.get(notification_template_type, []))
+            if status_str != 'running':
+                all_notification_templates.update(notification_templates.get('any', []))
```

Comment:
It looks like this behavior should go in the docs. If you attach a notification template as an "any" NT, that means that it'll send a notification as it completes. So it overlaps with success & failure, but not with start.

I see how that's sensible, but it's not obvious from the naming, and the docs should specifically enumerate when each type of notification template will be sent.

Reply:
Agreed.  I think this behavior makes the *most* sense, but it's also a little unclear that "all" really is more like "when the job finishes".

Reply:
I'll also say, I sort of hate this "any" feature because it's basically undocumented, doesn't exist in the UI, and you can accomplish it by just using "success" *and* "failure".  @wenottingham I think we should get rid of it.

Reply:
You'd need to migrate existing ones. But not against it.

Reply:
I'm also not against it, but if it happens, we should handle it as a separate pull request. For now, I think the main thing this needs before merging is content in the `docs/` folder & verification.

Reply:
New related AWX issue has been created here: https://github.com/ansible/awx/issues/4022